sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentField.cs

```diff
@@ -311,5 +326,23 @@ public AddressValue AsAddress()
 
             return ValueAddress;
         }
+
+        private object InternalValue => ValueType switch
```

Comment:
I worry this doesn't help enough with debugging.  Seeing a magic private field doesn't tell me how to go get the value myself in the way a `ToString` method that rendered as `[DocumentField: ValueType=String, AsString()=>Hello, World]` could.  Should we also make this a `public object GetValue()` method for anyone who wants to handle values generically?

Reply:
I think the private property approach improves debugging navigation since users won't need to call the conversion methods to access the field's internal value. On the other hand, I agree it may not solve the problem for users who have no idea on how to extract the value.

If we had a `public object GetValue()` method, wouldn't it be better to expose it as a `public object Value { get; }` directly? From my understanding, we added the conversion methods because users could struggle relying on casting, specially as we add more structured field values (such as `Currency`, `Address`, etc).

Reply:
Regarding the `ToString` implementation, what would be a good message for references? `AsCurrency`, for example, returns a `ValueCurrency`. Does `[DocumentField: ValueType=Currency, AsCurrency()=>Value]` sound reasonable?

Reply:
I like [DocumentField: ValueType=Currency, AsCurrency()=>Value] - where value is actually the result of the AsCurrency call. This seems like a good start and we can iterate over it as we get more feedback.

Reply:
I was actually thinking of literally writing out `Value`. If we print the actual internal value of the field, I think we could end up with a long string and not being too helpful. For example, for currency:
![image](https://user-images.githubusercontent.com/7576220/172695598-c7e77bcc-88f8-46c7-9373-76c2daef693a.png)

Or even for long strings:
![image](https://user-images.githubusercontent.com/7576220/172695402-427e8f70-e05b-450a-bd0b-0ac268f177f8.png)

Reply:
sounds good.